### PR TITLE
Be detailed card #2

### PR DESCRIPTION
**Pull request by uche-okoroafor**
### What this PR does (required):
 BE- detailed card   #2 
- Create methods for handling adding and removing sections of a card.

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**
Screenshot (140)
Screenshot (141)

Screenshot (142)

Screenshot (143)

Screenshot (144)

Screenshot (145)

Screenshot (146)

### Any information needed to test this feature (required):

- Use a PostMan app or Insomnia to make a request to the routes inside the "routes" directory.

- Also remove the " protect" function that checks for cookies in the address of the route.

### Any issues with the current functionality (optional):

- I had an issue with removing fields in the card, I was unable to find a MongoDB operation that can remove fields in a deep array, so I had to use Javascript to get the index and later applied it to a MongoDB operation and it worked, please if there is any better way to do it will be very good.

### CODE REVIEW

**Comment by ababber** [comment]
Good job so far Uche! I was looking at the schema for cards and columns, and I was wondering how you would link each card/task to each column? I'm in the middle of working on the drag and drop feature for cards and columns, and one of the things that set me back was not planning my local data template to accommodate which card is associated with which column, do you think we will need to do that on the back end as well?  For example, here is a local data template I am using on the front end:

`const defaultData: BoardData = {
  tasks: {
    'task-1': { id: 'task-1', title: 'task 1' },
  },
  uniqueTaskKey: 2,
  columns: {
    'column-1': {
      id: 'column-1',
      title: 'Not Started',
      taskIds: [''],
    },
    'column-2': {
      id: 'column-2',
      title: 'In Progress',
      taskIds: ['task-1'],
    },
    'column-3': {
      id: 'column-3',
      title: 'Completed',
      taskIds: [''],
    },
  },
  uniqueColumnKey: 4,
  columnOrder: ['column-1', 'column-2', 'column-3'],
};`

File: server/models/Card.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+const cardSchema = new mongoose.Schema({
+  cardTitle: {
+    type: String,
+    required: false
+  },
+  cardId: {
+    type: String,
+    required: false
+  },
```
Comment (moffatethan) [inline server/models/Card.js:11]: This doesn't need to be here, Mongo will generate an id on this by default.

File: server/controllers/checklist.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const { v4: uuidv4 } = require("uuid");
+
+exports.createChecklist = asyncHandler(async (req, res, next) => {
+  const { checklistItem, cardId, columnId, boardId, userId } = req.body;
+  if (!userId || !boardId || !columnId || !cardId || !checklistItem) {
+    res.status(400);
```
Comment (moffatethan) [inline server/controllers/checklist.js:8]: Hey, let's actually keep this here for when we implement our plugins. We can re-use this 😄

File: server/models/Card.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+const mongoose = require('mongoose')
+
+const cardSchema = new mongoose.Schema({
+  cardTitle: {
+    type: String,
+    required: false
```
Comment (moffatethan) [inline server/models/Card.js:6]: `required: false` is the default behaviour of Mongoose. We don't need to explicitly define it. Please change where applicable.

File: server/models/Card.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  attachment: {
+    type: String,
+    required: false
+  },
+  checklists: [
+    {
+      checklistId: {
+        type: String,
+        required: false
+      },
```
Comment (moffatethan) [inline server/models/Card.js:37]: If we're doing a reference store which we will with plugins this will not be needed.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Hey good work so far, 

My only concern right now is if the base of the branch is correct & we have no route for getting the card details. 

Also, as discussed in sprint planning we are going a different direction on the card actions which is fine to keep the code on check-lists as we can likely re-use that. 😄

File: server/config/cloudinaryConfig.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    cloudinary.uploader.upload(
+      file,
+      (result) => {
+        resolve({ url: result.url, id: result.public_id });
+      },
+      { folder: "MetaMorphic" },
+      { resource_type: "auto" }
+    );
+  });
+};
```
Comment (moffatethan) [inline server/config/cloudinaryConfig.js:20]: Not sure why this is here? It's not related to the detailed card work.

File: server/config/multerConfig.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+const multer = require("multer"),
+  path = require("path");
+//multer.diskStorage() creates a storage space for storing files.
+const imageStorage = multer.diskStorage({
+  destination: (req, file, cb) => {
+    if (file.mimetype === "image/jpeg" || file.mimetype === "image/png") {
```
Comment (moffatethan) [inline server/config/multerConfig.js:6]: Same here

File: server/controllers/card.js [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+exports.createChecklist = asyncHandler(async (req, res, next) => {
+  const { checklistItem, cardId, columnId, boardId, userId } = req.body;
+  const columnObjectId = ObjectID(columnId);
+  const boardObjectId = ObjectID(boardId);
+  const cardObjectId = ObjectID(cardId);
+  const createStatus = await User.updateOne(
+    { _id: userId, "boards.columns.cards._id": cardObjectId },
+    {
+      $push: {
+        "boards.$[board].columns.$[column].cards.$[card].checklists": {
```
Comment (moffatethan) [inline server/controllers/card.js:141]: As mentioned many times, this is actually not needed. We are doing a different approach so we can remove this code. Please make sure to be aware of what we're doing in the project going forward so we aren't spending time on unnecessary changes 👍🏻

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please remove dist files, unnecessary controllers for checklists and image uploads. Keep the cloudinary setup though so we have it. But please be aware of current discussions that we've had in standups 😄

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please address merge conflicts

**Review by moffatethan — APPROVED** [review]

**Pull request merged**